zcm/blocking.cpp

```diff
@@ -489,7 +489,7 @@ zcm_sub_t* zcm_blocking_t::subscribe(const string& channel,
     sub->usr = usr;
     sub->regex = isRegexChannel(channel);
     if (sub->regex) {
-        sub->regexobj = (void*) new std::regex(sub->channel);
+        sub->regexobj = (void*) new std::regex(channel);
```

Comment:
`channel` is the complete string.  `sub->channel` has been truncated at `ZCM_CHANNEL_MAXLEN`.  Note that just below here the sub is stored in a map using `channel` as the key.